icepyx/core/temporal.py

```diff
@@ -261,14 +271,18 @@ def validate_date_range_date(date_range, start_time=None, end_time=None):
     return _start, _end
 
 
-def validate_date_range_dict(date_range, start_time=None, end_time=None):
+def validate_date_range_dict(
+    date_range: dict[str, dt.datetime],
```

Comment:
```suggestion
    date_range: dict[str, dt.datetime, dt.date],
```

Reply:
dates and datetimes are related, so I think we can equivalently annotate `dict[str, dt.date]`:

```python
>>> import datetime as dt
>>> datetime = dt.datetime.now()
>>> date = dt.datetime.now().date()
>>> datetime
datetime.datetime(2024, 9, 25, 12, 47, 38, 861853)
>>> date
datetime.date(2024, 9, 25)
>>> isinstance(datetime, dt.date)
True
>>> isinstance(datetime, dt.datetime)
True
>>> isinstance(date, dt.date)
True
>>> isinstance(date, dt.datetime)
False
```

Reply:
d037e11

Reply:
Based on your description, it should be:
`dict[str, Union[str, dt.date]]` since str is also a valid input for the values.

Reply:
I see, thank you!

Reply:
97a5544

Reply:
https://github.com/icesat2py/icepyx/pull/604/files#r1777677428